lib/utils/database.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:path/path.dart';
+import 'package:sqflite/sqflite.dart';
+import 'dart:async';
+
+Future<Database> createDatabase() async {
+  final Future<Database> database = openDatabase(
+    join(await getDatabasesPath(), 'chirpnets.db'),
+    onCreate: (db, version) {
+      return db.execute(
+        """
+        CREATE TABLE users(id INTEGER PRIMARY KEY, name TEXT); 
+        CREATE TABLE groups(id INTEGER PRIMARY KEY, name TEXT);
+        CREATE TABLE messages(id INTEGER PRIMARY KEY, message TEXT);
```

Comment:
Are the messages going to be stored by conversation, or just all in one table mixed together?

Reply:
Looking at my in progress uml groups seems like what conversations would be, might be worth renaming?

Reply:
I dont know much about databases but wouldnt a table called conversations hold messages? Or will it hold references to the users table and references to the message table? would you remove the messages table and store in conversations table or should they be separate?